### PR TITLE
executor,infoschema: move `dataForUserPrivileges` from `infoschema` to `executor`

### PR DESCRIPTION
**Pull request by Rustin-Liu**
UCP #15038

Signed-off-by: Rustin-Liu <rustin.liu@gmail.com>

### What problem does this PR solve? 
#15038

### Check List 

Tests 

 - Unit test

Code changes
 - Has exported variable/fields change
 - Has persistent data change

Side effects

None

Related changes

None

Release note

None

### CODE REVIEW

**Comment by Rustin-Liu** [comment]
@sre-bot  /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot  /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot  /coverage

**Comment by reafans** [comment]
lgtm

**Comment by Rustin-Liu** [comment]
@lonng  PTAL~

**Comment by Rustin-Liu** [comment]
@sre-bot  /run-all-tests

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -385,3 +387,8 @@ func keyColumnUsageInTable(schema *model.DBInfo, table *model.TableInfo) [][]typ
 	}
 	return rows
 }
+
+func dataForUserPrivileges(ctx sessionctx.Context) [][]types.Datum {
```
Comment (zz-jason) [inline executor/infoschema_reader.go]: it's better to bind this function to `memtableRetriever`.

**Pull request merged**